src/Core/ServiceProvider.php

```diff
@@ -44,25 +43,13 @@ abstract class ServiceProvider extends AbstractServiceProvider implements Bootab
     /**
      * Get the container.
      *
-     * This method's actual return type and documented return type differ
-     * because PHP 7.2 doesn't support return type narrowing.
-     *
      * @return \Cake\Core\ContainerInterface
+     * @psalm-suppress MoreSpecificReturnType
```

Comment:
Don't know why psalm whines about covariant return type when it's supported by the language.

Reply:
I think the issue is the method types mismatch. The parent return guarantees only the base class while this return expects the derived class.

The types are allowed but the parent return isn't guaranteed to match.